HelloTorchIOS/HelloTorchIOS/LightsLogic.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  LightsLogic.swift
+//  HelloTorchIOS
+//
+//  Created by Luis Valle on 08/05/2018.
+//  Copyright © 2018 Novoda. All rights reserved.
+//
+
+import Foundation
+import AVFoundation
+
+class LightsLogic {
+    
+    private let device: AVCaptureDevice? = AVCaptureDevice.default(for: AVMediaType.video)
+    
+    init() {
+        do {
+            try device?.lockForConfiguration()
+        }catch {
+            print("Torch could not be used")
+        }
+    }
+    
+    func toggleStrobe() {
+        if (device?.hasTorch == false) {
```

Comment:
This could be a perfect use of a guard statement. A Guard statement on IOS is like an if/else but with the intent of returning out of the function if a condition is not met. 

```
guard device?.hasTorch else { 
  return 
} 
```